src/logics/SubstLoopBreaker.cc

```diff
@@ -205,37 +205,41 @@ std::vector<vec<SNRef>> SubstLoopBreaker::findLoops(vec<SNRef>& startNodes) {
     return loops;
 }
 
-MapWithKeys<PTRef,PtAsgn,PTRefHash> SubstLoopBreaker::constructLooplessSubstitution(MapWithKeys<PTRef,PtAsgn,PTRefHash>&& substs)
+Logic::SubstMap SubstLoopBreaker::constructLooplessSubstitution(MapWithKeys<PTRef,PtAsgn,PTRefHash>&& substs)
 {
+    Logic::SubstMap substPTRefs;
     for (PTRef key : substs.getKeys()) {
         PtAsgn & data = substs[key];
         if (data.sgn != l_True) {
             continue;
         }
 
         SNRef subst_node = sna.getSNRefBySource(key);
-        if (!sna[subst_node].hasChildren())
+        if (!sna[subst_node].hasChildren()) {
             data.sgn = l_False;
+        } else {
+            substPTRefs.insert(key, data.tr);
+        }
     }
-    return std::move(substs); // Since susbsts is a template, we need the move constructor here
+    return substPTRefs;
```

Comment:
I can't find any justification for the comment (I wrote in 2019-12-23) here.  I think it's not true.